controller/snapshot_controller.go

```diff
@@ -354,17 +361,27 @@ func (sc *SnapshotController) reconcile(snapshotName string) (err error) {
 		return nil
 	}
 
+	requestCreateNewSnapshot := snapshot.Spec.CreateSnapshot
+	alreadyCreatedBefore := snapshot.Status.CreationTime != ""
```

Comment:
`requestCreateNewSnapshot && alreadyCreatedBefore`: This case is a valid case, so it is possible

`!requestCreateNewSnapshot && !alreadyCreatedBefore`: This case is a buggy case. It will happen when user create a snapshot CR with `snapshot.Spec.CreateSnapshot: false`. This is a not valid operation because this type of snapshot should only be created by the engine controller.